src/telemetry/client.hpp

```diff
@@ -25,26 +25,29 @@
 #include "telemetry/signalhandler.hpp"
 #include "utils/logger.hpp"
 #include "utils/config.hpp"
+#include "client_interface.hpp"
 
 namespace hyped {
 
 using utils::Logger;
+using utils::Config;
+using utils::System;
 
 namespace telemetry {
 
-class Client {
+class Client : public ClientInterface {
   public:
-    explicit Client(Logger& log);
+    explicit Client();
```

Comment:
You don't need to use the explicit keyword here.

```suggestion
    Client();
```